### PR TITLE
Intel plugin template updates (RDT, PMU)

### PR DESCRIPTION
**Pull request by leifmadsen**
- Add Cores param to intel_rdt plugin
- Add intel_pmu plugin support
- Adjust test.yml

### CODE REVIEW

**Comment by leifmadsen** [comment]
[Add collectd templates to collectd_config ansible role](https://trello.com/c/8UHsT3CT/1105-add-collectd-templates-to-collectdconfig-ansible-role)

File: defaults/main/intel_pmu.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+# Ansible managed
```
Comment (elfiesmelfie) [inline defaults/main/intel_pmu.yml]: Was this generated with the template_generator? If so, the template file will be overwritten each time the generator is run. If it's just the defaults file that you kept from the generator, then you can remove the "# Ansible managed" string

File: templates/intel_pmu.conf.j2 [hunk 2]
```diff
@@ -0,0 +1,20 @@
+LoadPlugin intel_pmu
+
+<Plugin "intel_pmu">
+{% if collectd_plugin_intelpmu_cores is defined %}
+  Cores{% for item in collectd_plugin_intelpmu_cores %} "{{ item }}"{% endfor %}
```
Comment (elfiesmelfie) [inline templates/intel_pmu.conf.j2:5]: Is collectd_plugin_intelpmu_cores expecting a list, or a string?

Can you can a commented example in defaults/main/intel_pmu.yaml
e.g.
    collectd_plugin_intelpmu_dispatchmultipmu: false
    # collectd_plugin_intelpmu_cores: ["0", "2-3"]

File: templates/intel_pmu.conf.j2 [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+  ReportHardwareCacheEvents {{ collectd_plugin_intelpmu_reporthardwarecacheevents }}
+  ReportKernelPMUEvents {{ collectd_plugin_intelpmu_reportkernelpmuevents }}
+  ReportSoftwareEvents {{ collectd_plugin_intelpmu_reportsoftwareevents }}
+{% if collectd_plugin_intelpmu_eventlist is defined %}
+  EventList{% for item in collectd_plugin_intelpmu_eventlist %} "{{ item }}"{% endfor %}
+{% endif %}
+
+{% if collectd_plugin_intelpmu_hardwareevents is defined %}
+  HardwareEvents{% for item in collectd_plugin_intelpmu_hardwareevents %} "{{ item }}"{% endfor %}
```
Comment (elfiesmelfie) [inline templates/intel_pmu.conf.j2:16]: Same here with examples, I assume it's a list of strings

**Review by elfiesmelfie — COMMENTED** [review]
Commented inline, but I'll make the changes myself, and you can merge if it looks okay to you

**Review by elfiesmelfie — APPROVED** [review]

File: defaults/main/intel_pmu.yml (hunk 1 above)
Comment (leifmadsen) [reply defaults/main/intel_pmu.yml]: I'm still unsure what the expectation is here. Do you mean that I'd run a generator to get this template in the repository? Or are you referring to the file build configuration when running this in production?

Comment (elfiesmelfie) [reply defaults/main/intel_pmu.yml]: There's a script that creates the templates for the simple configs, it's found inhttps://github.com/infrawatch/osp-observability-ansible/tree/master/tools
I've been using is to create some of the simple configs and their defaults files.
In this case, the outputted templates and defaults files ( marked with "# Ansible managed" should not be modified manually.

At the moment, the template_generator doesn't reside in the same repo as this role, and may need to be rehomed.

Comment (leifmadsen) [reply defaults/main/intel_pmu.yml]: ah gotcha, so I don't actually want that manually added template header. I'll remove the header for now, and if we eventually end up with a generated version as part of the conversion work that's fine with me!

Comment (leifmadsen) [reply defaults/main/intel_pmu.yml]: Oh you already removed it :)

**Pull request merged**